scripts/zkai_d128_gate_value_compact_preprocessed_gate.py

```
@@ -0,0 +1,721 @@
+    writer.writerow(row)
+    write_bytes_atomic(path, handle.getvalue().encode("utf-8"))
+
+
+def write_bytes_atomic(path: pathlib.Path, data: bytes) -> None:
+    path.parent.mkdir(parents=True, exist_ok=True)
+    try:
+        metadata = path.lstat()
+    except FileNotFoundError:
+        metadata = None
+    if metadata is not None and pathlib.Path(path).is_symlink():
+        raise GateValueCompactGateError(f"refusing to overwrite symlink: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)